tests/test_read_user_profile.py

```diff
@@ -1,14 +1,21 @@
-import unittest
 import pathlib
-import d3a.constants
+import unittest
+
 from d3a_interface.constants_limits import GlobalConfig
 from d3a_interface.read_user_profile import copy_profile_to_multiple_days, \
     _read_from_different_sources_todict, time_str
+from pendulum import duration
+
+import d3a.constants
 from d3a.d3a_core.util import d3a_path
 
 
 class TestReadUserProfile(unittest.TestCase):
 
+    def setUp(self):
+        GlobalConfig.sim_duration = duration(days=1)
+        GlobalConfig.slot_length = duration(minutes=15)
+
```

Comment:
AFAIS we are resetting these values to their default, these most probably are altered in another test which is breaking this test
If this is True, it makes more sense if we tear down the altered value in the tests touching this variable
However this is minor and can be skipped, you have already made huge refactoring, Thanks!

Reply:
If @Hassan754 's comment is correct, this might fix this specific test but other tests might still break. I would also suggest changing this in the other test (it should be `test_strategy_pvpredefined`).

Reply:
I agree with you that this would definitely be the best way to go! Similar to the one-sided market configuration comment  though, I did not find the culprit test unfortunately, therefore I had to deal with the effect and not the cause of the issue :(

Reply:
So this doesn't depend on `test_strategy_pvpredefined`? I thought that was the only usage of these two constants in our tests. If not, let's leave it alone for the moment :)

Reply:
It did not depend only on this but on a couple more tests (test_simulation and test_live_events). Once I fixed these teardown methods the test succeeded, therefore this should not be needed anymore.